src/main/java/com/google/sps/data/Candidate.java

```diff
@@ -33,10 +34,31 @@ public static Builder builder() {
     return new AutoValue_Candidate.Builder();
   }
 
+  // creates a new Candidate object by extracting the properties from "obj"
+  public static Candidate fromJSONObject(JSONObject obj) throws JSONException {
+    return Candidate.builder()
+        .setName(obj.getString("name"))
+        .setPartyAffiliation(obj.getString("party"))
+
+        // TODO(caseyprice): get values for campaignSite and platformDescription
+        .setCampaignSite("")
+        .setPlatformDescription("")
+        .build();
+  }
+
+  // creates a new Entity and sets the proper properties.
+  public Entity toEntity() {
+    Entity entity = new Entity("Candidate");
+    entity.setProperty("name", this.getName());
+    entity.setProperty("partyAffiliation", this.getPartyAffiliation());
+    entity.setProperty("campaignSite", this.getCampaignSite());
+    entity.setProperty("platformDescription", this.getPlatformDescription());
+
+    return entity;
+  }
+
```

Comment:
Would an interface be useful for these Entity objects, where the names of functions are the same but implementation differs? I made an attempt at refactoring to have each of them implement an interface but decided against it because I didn't find it made any external implementation (in InfoCardServlet or ElectionServlet) easier.